src/core/opamSystem.mli

```diff
@@ -31,9 +31,13 @@ exception Internal_error of string
 (** Raise [Internal_error] *)
 val internal_error: ('a, unit, string, 'b) format4 -> 'a
 
-(** [with_tmp_dir fn] executes [fn] in a tempory directory *)
+(** [with_tmp_dir fn] executes [fn] creates a temporary directory and
+    passes its name to [fn]. The directory is alwasy removed on completion. *)
 val with_tmp_dir: (string -> 'a) -> 'a
 
+(** [in_tmp_dir fn] executes [fn] in a temporary directory. *)
+val in_tmp_dir: (unit -> 'a) -> 'a
```

Comment:
TIny thing, wondering if it should takes unit as argument or the temporary created directory name ?
```suggestion
val in_tmp_dir: (string -> 'a) -> 'a
```

Reply:
I wondered, but then I'd have had to do more than just use `in_dir` (and I didn't need the directory name) so I figured that if a function did need the actual directory then it could use `Sys.getcwd` 🙂